### PR TITLE
upgrate to jena-core 3.0.1, jena-arq 3.0.0, jena-tdb 3.0.0, jena-fuse…

### PR DESCRIPTION
**Pull request by flamingofugang**
hdt-fuseki/bin/hdtEndpoint.sh --config=<config file> can work now.

### CODE REVIEW

File: hdt-fuseki/bin/hdtEndpoint.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@ source `dirname $0`/javaenv.sh
 
 # Memory default settings
 INDEX_MEM=8g
-SERVER_MEM=1g
+SERVER_MEM=16g
```
Comment (bendiken) [inline hdt-fuseki/bin/hdtEndpoint.sh]: 16 gigs may be a bit much for a generic default setting, but let's see if anyone complains.

File: hdt-fuseki/pom.xml [hunk 2]
```diff
@@ -23,11 +23,12 @@
...
   	</dependency>
-      <dependency>
-        <groupId>org.apache.jena</groupId>
-        <artifactId>jena-fuseki1</artifactId>
-        <version>1.4.0</version>
-    </dependency>
+  	<dependency>
+  		<groupId>org.apache.jena</groupId>
+  		<artifactId>jena-fuseki1</artifactId>
+  		<version>1.3.0</version>
```
Comment (bendiken) [inline hdt-fuseki/pom.xml]: Is 1.3.0 instead of 1.4.0 necessary?

File: hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java [hunk 3]
```diff
@@ -101,7 +101,7 @@
     static {
         // Check if default command logging.
         if ( "set".equals(System.getProperty("log4j.configuration", "set") ) )
-            LogCtl.resetLogging(log4Jsetup) ;
+            LogCtl.resetLogging(log4Jsetup) ;
```
Comment (bendiken) [inline hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java]: When submitting patches please make sure to trim any stray trailing whitespace.

File: hdt-java-core/pom.xml [hunk 4]
```diff
@@ -30,7 +30,12 @@
 		<dependency>
 			<groupId>org.apache.jena</groupId>
 			<artifactId>jena-arq</artifactId>
-			<version>3.0.1</version>
+			<version>3.0.0</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.jena</groupId>
+			<artifactId>jena-tdb</artifactId>
```
Comment (bendiken) [inline hdt-java-core/pom.xml]: Is Jena TDB really a necessary dependency?

File: hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java [hunk 5]
```diff
@@ -149,9 +149,9 @@
 
     static public void main(String...argv)
     {
-        // Just to make sure ...
-        ARQ.init() ;
-        TDB.init() ;
+        // // Just to make sure ...
+        // ARQ.init() ;
+        // TDB.init() ;
         Fuseki.init() ;
```
Comment (bendiken) [inline hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java]: Are these changes intentional? If Fuseki calls `ARQ.init()`, then it'd be better to remove this line altogether here, instead of commenting it out.

File: hdt-fuseki/config-hdt.ttl [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    fuseki:dataset                   <#dataset_pubchemrdf> ;
+    .
+
+
+<#dataset_pubchemrdf> rdf:type ja:RDFDataset ;
+    ja:namedGraph 
+        [ ja:graphName  <http://rdf.ncbi.nlm.nih.gov/pubchem/compound> ;
+          ja:graph      <#graph_compound> ] ,
+	    [ ja:graphName  <http://rdf.ncbi.nlm.nih.gov/pubchem/substance> ;
+          ja:graph      <#graph_substance> ] ,
```
Comment (bendiken) [inline hdt-fuseki/config-hdt.ttl]: This file is awfully specific to NCBI. Did you really mean to import it into the code base?

File: hdt-java-core/pom.xml (hunk 4 above)
Comment (flamingofugang) [inline hdt-java-core/pom.xml]: Yes, it is required to run hdt-fuseki/bin/hdtEndpoint.sh binary.

Best,
Gang

File: hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java (hunk 3 above)
Comment (flamingofugang) [inline hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java]: Sure, will do next time.

File: hdt-fuseki/config-hdt.ttl (hunk 6 above)
Comment (flamingofugang) [inline hdt-fuseki/config-hdt.ttl]: Sorry, I forgot to remove it. You can ignore this one.

File: hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java (hunk 5 above)
Comment (flamingofugang) [inline hdt-fuseki/src/main/java/org/rdfhdt/hdt/fuseki/FusekiHDTCmd.java]: Yes, otherwise, it won’t run. Arq.init() and tdb.init() are old functions and depend on old packages that are not part of dependency. If you look at the newest verion of FusekiCmd.java on github, you will see they are excluded there.

I attempted to comment them out, but you can actually delete them, I suppose it won’t be used any more in the future.

File: hdt-fuseki/pom.xml (hunk 2 above)
Comment (flamingofugang) [inline hdt-fuseki/pom.xml]: Yes, to be honest, I have tried all of them, only 1.3.0 can work. The current codes are really old, they usually refer to some old packages which are most of time missing in the dependency packages….

File: hdt-fuseki/bin/hdtEndpoint.sh (hunk 1 above)
Comment (flamingofugang) [inline hdt-fuseki/bin/hdtEndpoint.sh]: It is setup to run PubChemRDF data, which contains billions of tripes. It is not necessary to merge to base.

Comment (bendiken) [inline hdt-fuseki/bin/hdtEndpoint.sh]: Note that I can't really selectively ignore things in pull requests, so everything gets merged. You can, however, add more commits to pull requests (for the future) to change things prior to merges.

Comment (flamingofugang) [inline hdt-fuseki/bin/hdtEndpoint.sh]: I see, will do in the future.

**Comment by bendiken** [comment]
@Callidon, @hinahiroy Please review as well.

**Comment by bendiken** [comment]
@flamingofugang I've removed the stray configuration file in 0d7ae65c4f1ed659b8f1f16181577fd5bec053f4. Please take a backup of your local file before pulling, as it will get deleted when pulling.

File: hdt-java-core/pom.xml (hunk 4 above)
Comment (Callidon) [inline hdt-java-core/pom.xml]: I doesn't agree with this. As `jena-tdb` is include as a dependency of `jena-fuseki1` in *hdt-fuseki*, there is no need to use it in *hdt-java-core*. To be sure, I used the [Maven dependency plugin](http://maven.apache.org/plugins/maven-dependency-plugin/) to check for unused dependencies and it reported `jena-tdb` as unused :
```
[INFO] ------------------------------------------------------------------------
[INFO] Building HDT Java Core 2.0
[INFO] ------------------------------------------------------------------------
...
[INFO] --- maven-dependency-plugin:2.10:analyze (default-cli) @ hdt-java-core ---
[WARNING] Used undeclared dependencies found:
[WARNING]    junit:junit:jar:4.11:compile
[WARNING]    org.apache.jena:jena-core:jar:3.0.0:compile
[WARNING]    org.apache.jena:jena-base:jar:3.0.0:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.apache.jena:jena-tdb:jar:3.0.0:compile
[WARNING]    junit:junit-dep:jar:4.11:compile
[WARNING]    com.beust:jcommander:jar:1.32:compile
[WARNING]    com.github.jsonld-java:jsonld-java:jar:0.7.0:compile
```

Comment (bendiken) [inline hdt-java-core/pom.xml]: If the dependency is not needed, we should definitely drop it.

Comment (Callidon) [inline hdt-java-core/pom.xml]: I try and the hdt-fuseki/bin/hdtEndpoint.sh binary can be run without this dependency. I also noticed a bunch of unsused and undeclared dependencies in other modules of the project. If you want, I can make another pull request to clean the pom.xml files.

Comment (bendiken) [inline hdt-java-core/pom.xml]: That would be most welcome, @Callidon.
